### PR TITLE
Add a new image and distro definition for using on the WebKit CI for WPE perf bots

### PR DESCRIPTION
**Pull request by clopez**
* This adds a new image named `core-image-webkit-dev-ci-tools-tests` which will be used on the WebKit CI for building an image containing everything needed to build WebKit and run tests (perf-tests, layout-tests, etc). The image itself doesn't contain WebKit (WPE).

* It adds also a `python-is-python3` recipe that creates a symlink to provide the unversioned python interpreter as python3.

For more info see: https://bugs.webkit.org/show_bug.cgi?id=249604

Note: this patch is also on the WebKit PR at https://github.com/WebKit/WebKit/pull/7885 (file: [Tools/yocto/rpi/fix-nativesdk-image-defs.patch](https://github.com/WebKit/WebKit/blob/de647a4d3c36d2dd6923a8ac0ac653825c7790b5/Tools/yocto/rpi/fix-nativesdk-image-defs.patch))
The idea is to update that WebKit PR (or do a new PR if that one lands before this) with the reduced delta.

### CODE REVIEW

**Comment by clopez** [comment]
This image definition is also used to build a cross-toolchain and SDK with:

```
bitbake -c populate_sdk core-image-webkit-dev-ci-tools-tests
```

File: recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb [hunk 1]
```diff
@@ -254,31 +264,26 @@ RDEPENDS:packagegroup-wpewebkit-depends-tests = " \
     curl \
     gdb \
     php \
-    ${@bb.utils.contains('BBFILE_COLLECTIONS', 'meta-python2', 'python python-subprocess32 python-pygobject python-psutil', '', d)} \
+    ${@bb.utils.contains('BBFILE_COLLECTIONS', 'meta-python2', 'python python-subprocess32 python-pygobject python-psutil', 'python-is-python3', d)} \
```
Comment (psaavedra) [inline recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:267]: Since python2 is not longer a requirement (https://docs.yoctoproject.org/migration-guides/migration-3.1.html). I'd suggest to remove the meta-python2 and any kind of reference to it.

We can do it in a separate MR.

Comment (psaavedra) [reply recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:267]: Added it https://github.com/Igalia/meta-webkit/pull/437/commits/2b2b841473b3b809a0f1b7478d7d7a7509fabc9c ( https://github.com/Igalia/meta-webkit/pull/437/commits )

File: recipes-devtools/python/python-is-python3.bb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+DESCRIPTION = "Convenience package which ships a symlink to point the python interpreter to python3"
```
Comment (psaavedra) [inline recipes-devtools/python/python-is-python3.bb:1]: is this actually needed if you don use `meta-python2` layer?

**Review by psaavedra — CHANGES_REQUESTED** [review]

File: recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb [hunk 3]
```diff
@@ -0,0 +1,122 @@
+SUMMARY = "Image with all the required tools to build WPE (using the script build-webkit) \
```
Comment (psaavedra) [inline recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb:1]: Could I suggest a shorter image name: `s/core-image-webkit-dev-ci-tools-tests/`**`webkit-dev-ci-tools`**`/` ?

File: recipes-devtools/python/python-is-python3.bb (hunk 2 above)
Comment (clopez) [reply recipes-devtools/python/python-is-python3.bb:1]: It is good to have it because several python scripts inside the WebKit Tools directory still have python as shebang instead of python3. As far as I tested they work fine with python3 as long as you have the symlink.

One example is the kraken benchmark script run_benchmark. It ends calling  this:
`Tools/Scripts/webkitpy/benchmark_runner/data/plans/kraken.plan:    "create_script": ["python", "make-hosted.py"],`

File: recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb (hunk 3 above)
Comment (clopez) [reply recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb:1]: Sounds fine, I will change it.

File: conf/distro/webkitdevci.conf [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Distro based on Poky
+require conf/distro/poky.conf
+
+DISTRO = "webkitdevci"
+DISTRO_NAME= "WebKit Dev@CI (Yocto Poky Based Distro)"
+
+# Expand inmediately (:=) this two variables as soon as they are assigned (now),
+# otherwise the (default) lazy expansion causes issues with BASEFILESISSUEINSTALL
+# at recipe 'base-files' due to 'metadata not deterministic' errors.
+WEBKIT_CROSS_TARGET := "${@d.getVar('BB_ORIGENV').getVar('WEBKIT_CROSS_TARGET') or '${MACHINE}'}"
```
[the remark this answers is not in the document]
Comment (clopez) [reply conf/distro/webkitdevci.conf:10]: sure, I will document that

File: recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb [hunk 5]
```diff
@@ -133,7 +135,7 @@ RDEPENDS:packagegroup-wpewebkit-depends-core = "\
     sqlite3 \
     zlib \
     libpng \
-    libsoup-2.4 \
+    libsoup \
```
[the remark this answers is not in the document]
Comment (clopez) [reply recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:138]: great catch. I added it the new version of the patch

File: recipes-devtools/python/python-is-python3.bb (hunk 2 above)
Comment (psaavedra) [reply recipes-devtools/python/python-is-python3.bb:1]: ok. understood.

File: recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb (hunk 1 above)
Comment (psaavedra) [reply recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:267]: https://github.com/Igalia/meta-webkit/pull/437/commits merged.

File: recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb (hunk 5 above)
[the remark this answers is not in the document]
Comment (psaavedra) [reply recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:138]: good

File: recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb (hunk 1 above)
Comment (clopez) [reply recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb:267]: good riddance, i will remove references to meta-python2 as well

**Comment by clopez** [comment]
I'm uploading a new version of the patch that I think address the previous comments and also tunes a bit further how the image is generated to also include -dev packages and so it becomes possible to actually build WebKit on the target device (native build) if wished.

File: conf/distro/webkitdevci.conf [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+# Distro features
+DISTRO_FEATURES:append = " opengl egl pam systemd"
+DISTRO_FEATURES:remove = "ptest"
+DISTRO_FEATURES_NATIVESDK:append = " wayland"
+
+# Use systemd
+DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
+VIRTUAL-RUNTIME_init_manager = "systemd"
+VIRTUAL-RUNTIME_dev_manager = "systemd"
```
Comment (psaavedra) [inline conf/distro/webkitdevci.conf:36]: Could I suggest the addition of TUI for the GDB?:

```suggestion
VIRTUAL-RUNTIME_dev_manager = "systemd"

# Add the GDB Text User Interface (TUI) terminal interface
PACKAGECONFIG:append:pn-gdb = " tui"
PACKAGECONFIG:append:pn-gdb-cross = " tui"
PACKAGECONFIG:append:pn-gdb-cross-canadian = " tui"
```

**Review by psaavedra — APPROVED** [review]
Other than the GDB TUI suggestion. The PR looks good. r+.

**Comment by clopez** [comment]
Committed it to main on 46fbfbc58f82abc3dc8794c630172d4554be0830
I added the GDB TUI suggestion and also a few other packageconfig options that were previously set on the targets local.conf (in the WebKit patch) but makes more sense to have those options here in the distro definition
